src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2619,5 +2639,20 @@ void InterpreterSelectQuery::initSettings()
 
     }
 }
+Chunk InterpreterSelectQuery::create_single_cache_chunk_from_many(IAST::Hash ASTHash)
```

Comment:
This method will not work, chunk has multiple columns, and when you merge chunks you need to merge each column independently. Assume we have 2 chunks with 2 columns, then during merge we will have 1 chunk with 2 columns and each column will contain data from 2 chunks. Check `IColumn.h` and `insertFrom` methods.

Reply:
Fixed.